SmartDeviceLink/SDLFileManager.m

```diff
@@ -260,6 +260,20 @@ - (void)uploadFiles:(NSArray<SDLFile *> *)files completionHandler:(nullable SDLF
     [self uploadFiles:files progressHandler:nil completionHandler:completionHandler];
 }
 
+- (void)uploadArtwork:(SDLArtwork *)artwork completionHandler:(nullable SDLFileManagerUploadArtworkCompletionHandler)completion {
+    if ([self.remoteFileNames containsObject:artwork.name] && !artwork.overwrite) {
+        // Artwork with same name already uploaded to remote; return artwork name
+        if (completion == nil) { return; }
+        return completion(true, artwork.name, self.bytesAvailable, nil);
+    }
+
+    // Upload artwork and return artwork name when finished
+    [self uploadFile:artwork completionHandler:^(BOOL success, NSUInteger bytesAvailable, NSError * _Nullable error) {
```

Comment:
It appears that the `uploadFile` method doesn't take into account the `uploadedEphemeralFiles`, I think bugs may occur with this method as a result. Only `uploadFiles` seems to take it into account. Perhaps that code should be moved to `uploadFile`.

Reply:
Fixed. I moved the `uploadedEphemeralFiles` check to `uploadFile:` from `uploadFiles:`. I also fixed & added test cases related to `uploadedEphemeralFiles`.